src/app/shared/components/disk-offering/disk-offering-selector/disk-offering-selector.component.ts

```diff
@@ -73,4 +109,13 @@ export class DiskOfferingSelectorComponent implements ControlValueAccessor {
         }
       });
   }
+
+  private setMaxSizeValue() {
+    const customDiskOfferingMaxSize = this.authService.getCustomDiskOfferingMaxSize();
+    this.min = this.min ? this.min : this.authService.getCustomDiskOfferingMinSize();
+    this.max =
+      !this.availableStorage && isNaN(Number(this.availableStorage))
```

Comment:
Why do you use both `!this.availableStorage` and `isNaN(Number(this.availableStorage))`? Can't you use just isNaN?